### PR TITLE
HHVM was removed. Remove from travis conditionals

### PR DESCRIPTION
**Pull request by photodude**
Pull Request for Issue complete removal started with https://github.com/joomla/joomla-cms/pull/16232

### Summary of Changes
removes some travis items that were specific to HHVM

### Testing Instructions
review

### Expected result

unit tests pass

### Actual result
unit tests pass

### Documentation Changes Required
HHVM might work but is unsupported and no longer tested (add notice in requirements if not existing)

### Additional information
in addition to passing our unit tests, we did have [one user who did local testing on hhvm after the unit test were passing](https://groups.google.com/d/msg/joomla-dev-cms/R0_ARiHuzOI/nKPgWCBRDQAJ) 

Basically, Joomla CMS may work on HHVM but not supported. (kind of like mariaDB and other mysql variants)

There are 2 major reasons that Laravel, [Symfony](http://symfony.com/blog/symfony-4-end-of-hhvm-support), [Doctrine](https://twitter.com/lcobucci/status/865081210128793601), CakePHP, MongoDB, PHPUnit, Composer, Twig, Silex, and Swiftmailer, etc have all dropped support for HHVM (that is something like 75% of the php market use has dropped support). 1) HHVM has needed to fix some big compatibility issues and 2) HHVM needs to improve their support when library devs report issues. With most of the big libraries moving to PHP7 only issues like [HHVM and PHP7 differ on type annotations on internal functions (rtrim in particular) hhvm 7198](https://github.com/facebook/hhvm/issues/7198) block the ability to test HHVM in [HHVM's PHP7 mode](https://docs.hhvm.com/hhvm/configuration/INI-settings#php-7-settings) since composer is unable to function due to that bug. Additionally, Cpanel and the like have never placed HHVM into the standard hosting options with PHP; resulting in limited user adoption with only places like Etsy on VMs or dedicated servers using HHVM. Add to that Travis testing of HHVM has tended to be outdated or slow and the new trusty container on travis is broken with mysql; testing is often the first step to support. 

Personally, I hope that HHVM gets their act together so we and others can consider supporting HHVM again but at the moment it's a lame duck with little progress for those not programming with hiphop.

### CODE REVIEW

File: build/travis/unit-tests.sh [hunk 1]
```diff
@@ -35,5 +29,5 @@ if [[ $INSTALL_APC == "yes" ]]; then phpenv config-add "$BASE/build/travis/phpen
 if [[ $INSTALL_APCU == "yes" && $TRAVIS_PHP_VERSION = 5.* ]]; then printf "\n" | pecl install apcu-4.0.10 && phpenv config-add "$BASE/build/travis/phpenv/apcu-$TRAVIS_PHP_VERSION.ini"; fi
 if [[ $INSTALL_APCU == "yes" && $TRAVIS_PHP_VERSION = 7.* ]]; then printf "\n" | pecl install apcu-beta && phpenv config-add "$BASE/build/travis/phpenv/apcu-$TRAVIS_PHP_VERSION.ini"; fi
 if [[ $INSTALL_APCU_BC_BETA == "yes" ]]; then printf "\n" | pecl install apcu_bc-beta; fi
-if [[ $INSTALL_REDIS == "yes" && $TRAVIS_PHP_VERSION != hhvm ]]; then phpenv config-add "$BASE/build/travis/phpenv/redis.ini"; fi
-if [[ $INSTALL_REDIS == "yes" && $TRAVIS_PHP_VERSION = hhvm ]]; then cat "$BASE/build/travis/phpenv/redis.ini" >> /etc/hhvm/php.ini; fi
+if [[ $INSTALL_REDIS == "yes" ]]; then phpenv config-add "$BASE/build/travis/phpenv/redis.ini"; fi
+if [[ $INSTALL_REDIS == "yes" ]]; then cat "$BASE/build/travis/phpenv/redis.ini" >> /etc/hhvm/php.ini; fi
```
Comment (Quy) [inline build/travis/unit-tests.sh]: Remove `/etc/hhvm/php.ini`?

Comment (Quy) [inline build/travis/unit-tests.sh]: Also lines 32 and 33 have the same tests. Combine them?

Comment (rdeutz) [inline build/travis/unit-tests.sh]: both lines can be removed

Comment (rdeutz) [inline build/travis/unit-tests.sh]: ah only the second

Comment (photodude) [inline build/travis/unit-tests.sh]: Oversight should have been removed. completed now.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7f3dab47a6da573a85bec974794e521d55a6ff0f<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16281">issues.joomla.org/tracker/joomla-cms/16281</a>.</sub>

**Review by yvesh — APPROVED** [review]

**Pull request merged**